app/src/app/views/stix/relationship/relationship-view/relationship-view.component.ts

```diff
@@ -34,20 +34,35 @@ export class RelationshipViewComponent extends StixViewPage implements OnInit {
 
     ngOnInit(): void {
         // initialize source/target types if there is a source/target object, or if there is only one possible value
-        if (this.relationship.source_object) this.source_type = stixTypeToAttackType[this.relationship.source_object.stix.type]
+        if (this.relationship.source_object) {
+            this.source_type = stixTypeToAttackType[this.relationship.source_object.stix.type]
+            // retrieve parent of source object
+            var src_sub = this.relationship.get_parent_object(this.relationship.source_object, this.restApiService).subscribe({
+                next: (res) => { this.relationship.source_parent = res; },
+                complete: () => { if (src_sub) src_sub.unsubscribe(); }
+            });
```

Comment:
This assignment logic should probably be contained in the relationship itself as a new function `update_parent(restApiService): Observable<Relationship>`.